src/dagmc/tools/test_geom.cpp

```diff
@@ -0,0 +1,1158 @@
+  }
+  // cube0
+  for (unsigned i = 0; i < num_tris / 2; ++i) {
+    const EntityHandle conn[] = { verts[connectivity[3 * i  ]],
+                                  verts[connectivity[3 * i + 1]],
+                                  verts[connectivity[3 * i + 2]]
+                                };
+    rval = moab->create_element(MBTRI, conn, 3, tris[i]);
+    CHKERR;
+  }
+  // cube1
+  for (unsigned i = 0; i < num_tris / 2; ++i) {
+    const EntityHandle conn[] = { verts[8 + connectivity[3 * i  ]],
+                                  verts[8 + connectivity[3 * i + 1]],
+                                  verts[8 + connectivity[3 * i + 2]]
+                                };
+    rval = moab->create_element(MBTRI, conn, 3, tris[num_tris / 2 + i]);
+    CHKERR;
+  }
+
+  // create CAD topology
+  EntityHandle* tri_iter = tris;
+  for (unsigned i = 0; i < num_surfs; ++i) {
+    rval = moab->create_meshset(MESHSET_SET, surfs[i]);
+    CHKERR;
+    rval = moab->add_entities(surfs[i], tri_iter, tris_per_surf);
+    CHKERR;
+    tri_iter += tris_per_surf;
+  }
+
+  Tag dim_tag, id_tag, sense_tag;
+  rval = moab->tag_get_handle(GEOM_DIMENSION_TAG_NAME,
+                              1, MB_TYPE_INTEGER,
+                              dim_tag,
+                              MB_TAG_SPARSE | MB_TAG_CREAT);
+  CHKERR;
+  rval = moab->tag_get_handle(GLOBAL_ID_TAG_NAME,
+                              1, MB_TYPE_INTEGER,
+                              id_tag,
+                              MB_TAG_DENSE | MB_TAG_CREAT);
+  CHKERR;
+  rval = moab->tag_get_handle("GEOM_SENSE_2",
+                              2, MB_TYPE_HANDLE,
+                              sense_tag,
+                              MB_TAG_SPARSE | MB_TAG_CREAT);
+  CHKERR;
+
+  std::vector<int> dims(num_surfs, 2);
+  rval = moab->tag_set_data(dim_tag, surfs, num_surfs, &dims[0]);
+  CHKERR;
+  std::vector<int> ids(num_surfs);
+  for (size_t i = 0; i < ids.size(); ++i)
+    ids[i] = i + 1;
+  rval = moab->tag_set_data(id_tag, surfs, num_surfs, &ids[0]);
+  CHKERR;
+
+  EntityHandle volume;
+  rval = moab->create_meshset(MESHSET_SET, volume);
+  CHKERR;
+  for (unsigned i = 0; i < num_surfs; ++i) {
+    rval = moab->add_parent_child(volume, surfs[i]);
+    CHKERR;
+  }
+
+  std::vector<EntityHandle> senses(2 * num_surfs, 0);
+  for (size_t i = 0; i < senses.size(); i += 2)
+    senses[i] = volume;
+  rval = moab->tag_set_data(sense_tag, surfs, num_surfs, &senses[0]);
+  CHKERR;
+
+  const int three = 3;
+  const int one   = 1;
+  rval = moab->tag_set_data(dim_tag, &volume, 1, &three);
+  CHKERR;
+  rval = moab->tag_set_data(id_tag, &volume, 1, &one);
+  CHKERR;
+
+  rval = moab->write_mesh(output_file_name);
+  CHKERR;
+  delete moab;
+  return MB_SUCCESS;
+}
+
+static bool run_test(std::string name, int argc, char* argv[]) {
+  if (argc == 1)
+    return true;
+  for (int i = 1; i < argc; ++i)
+    if (name == argv[i])
+      return true;
+  return false;
+}
+
+#define RUN_TEST(A) do { \
```

Comment:
Looks like you have a `do` / `while(false)` loop in this macro that is effectively inert other than a brace scope change?  If the purpose of this is to just create a brace level, you should be able to do that with just empty braces.